projects/fsastorefrontlib/src/cms-components/sales-illustration/components/sales-illustration-main/sales-illustration-main.component.ts

```diff
@@ -0,0 +1,272 @@
+  OnDestroy,
+  OnInit,
+  QueryList,
+  ViewChildren,
+} from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { NgbNav } from '@ng-bootstrap/ng-bootstrap';
+import { PaginationModel, RoutingService } from '@spartacus/core';
+import {
+  FormDataService,
+  FormDataStorageService,
+} from '@spartacus/dynamicforms';
+import { BREAKPOINT, BreakpointService } from '@spartacus/storefront';
+import { ECharts, EChartsOption } from 'echarts';
+import {
+  BehaviorSubject,
+  combineLatest,
+  fromEvent,
+  Observable,
+  of,
+  ReplaySubject,
+} from 'rxjs';
+import {
+  debounceTime,
+  filter,
+  first,
+  map,
+  shareReplay,
+  switchMap,
+  takeUntil,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { FSCartService, FSProductService } from '../../../../core';
+import { PricingService } from '../../../../core/product-pricing/facade/pricing.service';
+import { PricingData } from '../../../../occ/occ-models/form-pricing.interface';
+import { PaginationHelper } from '../../../../shared/util/helpers/pagination/PaginationHelper';
+import { logger } from '../../../../shared/util/helpers/rxjs/logger';
+import { ChartService } from '../../services/chart.service';
+
+@Component({
+  selector: 'cx-fs-sales-illustration-main',
+  templateUrl: './sales-illustration-main.component.html',
+})
+export class SalesIllustrationMainComponent
+  implements OnInit, AfterViewInit, OnDestroy {
+  @ViewChildren('nav') nav: QueryList<NgbNav>;
+
+  private destroyed$: ReplaySubject<boolean> = new ReplaySubject();
+
+  pricingData$: Observable<PricingData>;
+
+  productCode$: Observable<string>;
+  product$: Observable<any>;
+  tableData$: Observable<any>;
+
+  eChartsInstance: ECharts;
+  chartOptions$: Observable<EChartsOption>;
+
+  pagination$: BehaviorSubject<PaginationModel> = new BehaviorSubject({
+    currentPage: 0,
+    pageSize: 5,
+    totalPages: 0,
+    totalResults: 0,
+  });
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private productService: FSProductService,
+    private formDataStorageService: FormDataStorageService,
+    private formDataService: FormDataService,
+    private pricingService: PricingService,
+    private chartService: ChartService,
+    private cartService: FSCartService,
+    private routingService: RoutingService,
+    private breakpointService: BreakpointService,
+    public location: Location
+  ) {}
+
+  ngAfterViewInit(): void {
+    combineLatest([this.breakpointService.breakpoint$, this.nav.changes])
+      .pipe(
+        tap(([breakpoint, change]) => {
+          breakpoint === BREAKPOINT.xs ? change.first.select(2) : null;
+        }),
+        takeUntil(this.destroyed$)
+      )
+      .subscribe();
+  }
+
+  ngOnInit(): void {
+    fromEvent(window, 'resize')
+      .pipe(
+        debounceTime(300),
+        tap(e => this.resizeChart()),
+        takeUntil(this.destroyed$)
+      )
+      .subscribe();
+
+    this.productCode$ = this.activatedRoute.params.pipe(
```

Comment:
Could this have been initialized immediately after declaration, in the same line?
Do we need this anyway? If it's only for debugging it should be removed, the helper as well.

Reply:
It's needed, without it I can't get product

Reply:
> It's needed, without it I can't get product

Yes but do we need `logger('Product Code')` inside it?